src/test/txrebroadcast_tests.cpp

```diff
@@ -57,6 +57,48 @@ BOOST_AUTO_TEST_CASE(recency)
     BOOST_CHECK_EQUAL(candidates.front().m_txid, tx_old.GetHash());
 }
 
+BOOST_AUTO_TEST_CASE(fee_rate)
+{
+    // Since the test chain comes with 100 blocks, the first coinbase is
+    // already valid to spend. Generate another block to have two valid
+    // coinbase inputs to spend.
+    CreateAndProcessBlock(std::vector<CMutableTransaction>(), CScript());
+
+    // Instantiate rebroadcast module & mine a block, so when we run
+    // GetRebroadcastTransactions, Chain tip will be beyond m_tip_at_cache_time
+    const auto chain_params = CreateChainParams(*m_node.args, CBaseChainParams::MAIN);
+    TxRebroadcastHandler tx_rebroadcast(*m_node.mempool, *m_node.chainman, *chain_params);
+    CBlock recent_block = CreateAndProcessBlock(std::vector<CMutableTransaction>(), CScript());
+    CBlockIndex recent_block_index{recent_block.GetBlockHeader()};
+
+    // Update m_cached_fee_rate
+    // The transactions created in this test are each 157 bytes, and they set
+    // the fee at 1 BTC and 2 BTC.
+    CFeeRate cached_fee_rate(1.5 * COIN, 157);
+    tx_rebroadcast.UpdateCachedFeeRate(cached_fee_rate);
+
+    // Create two transactions
+    CKey key;
+    key.MakeNewKey(true);
+    CScript output_destination = GetScriptForDestination(PKHash(key.GetPubKey()));
+
+    // - One with a low fee rate
+    CMutableTransaction tx_low = CreateValidMempoolTransaction(m_coinbase_txns[0], /* vout */ 0, /* input_height */ 0, coinbaseKey, output_destination, CAmount(49 * COIN));
+    // - One with a high fee rate
+    CMutableTransaction tx_high = CreateValidMempoolTransaction(m_coinbase_txns[1], /* vout */ 0, /* input_height */ 1, coinbaseKey, output_destination, CAmount(48 * COIN));
+
+    // Confirm both transactions successfully made it into the mempool
+    BOOST_CHECK_EQUAL(m_node.mempool->size(), 2);
+
+    // Age transaction to be older than REBROADCAST_MIN_TX_AGE
+    SetMockTime(GetTime<std::chrono::seconds>() + 35min);
```

Comment:
In 0ae1420767a320cfeafd2acee0b54bc6a6eb9b27 [test] Add unit test for the fee rate cache 

Why note just use + REBROADCAST_MIN_TX_AGE + 5 minutes here?

Reply:
`REBROADCAST_MIN_TX_AGE` is defined in the `.cpp`, so isn't available to import in the test. I could move it to the header, but I was trying to minimize how much stuff is in the header only for tests. I could redefine the symbol as a constant in the test, but then we run into the same issue of having to update it if the code changes, so doesn't seem much better..